src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -0,0 +1,470 @@
+            {
+                final Colony colony = getColony();
+                if(colony != null)
+                {
+                    final AbstractBuilding building = colony.getBuilding(pos);
+                    if(building instanceof BuildingDeliveryman)
+                    {
+                        return ((BuildingDeliveryman) building).getBuildingToDeliver() != null
+                                && ((BuildingDeliveryman) building).getBuildingToDeliver().getLocation() == buildingEntry.getLocation();
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check all chests in the worker hut for a required item.
+     * @param is the type of item requested (amount is ignored)
+     * @return true if a stack of that type was found
+     */
+    public boolean isInHut(@Nullable final ItemStack is)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+        if(building != null)
+        {
+            if(isInTileEntity(building.getTileEntity(), is))
+            {
+                return true;
+            }
+
+            for(final BlockPos pos : building.getAdditionalCountainers())
+            {
+                @Nullable final TileEntity entity = worldObj.getTileEntity(pos);
+                if(entity instanceof TileEntityChest && isInTileEntity((TileEntityChest) entity, is))
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Check for a certain item and return the position of the chest containing it.
+     * @param stack the stack to search for.
+     * @return the position or null.
+     */
+    @Nullable
+    public BlockPos getPositionOfChestWithItemStack(ItemStack stack)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        if(building != null)
+        {
+            if(isInTileEntity(building.getTileEntity(), stack))
+            {
+                return building.getLocation();
+            }
+
+            for(final BlockPos pos : building.getAdditionalCountainers())
+            {
+                final TileEntity entity = worldObj.getTileEntity(pos);
+                if(entity instanceof TileEntityChest && isInTileEntity((TileEntityChest) entity, stack))
+                {
+                    return pos;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Check for a certain item and return the position of the chest containing it.
+     * @param tool the tool to search for.
+     * @param minLevel the minLevel of the pickaxe
+     * @param requestingBuilding the building requesting it.
+     * @return the position or null.
+     */
+    public BlockPos getPositionOfChestWithTool(@NotNull final String tool,final int minLevel, @NotNull final AbstractBuilding requestingBuilding)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getPositionOfChestWithTool" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)